examples/favorites_advanced/bloc_favorites_advanced/lib/feature_puppy/edit/bloc/puppy_edit_form_bloc.dart

```diff
@@ -0,0 +1,252 @@
+  PuppyEditFormBloc({required this.repository,required Puppy puppy})
+      : _puppy = puppy,
+        // puppy1 = puppy,
+        super(autoValidate: false, isLoading: true) {
+    addFieldBlocs(
+      fieldBlocs: [
+        // isSaveEnabled,
+        name,
+        breed,
+        gender,
+        characteristics,
+      ],
+    );
+
+
+    name.onValueChanges(onData: (previous, current) async* {
+      if (previous != current) {
+        // print('name current value: ${current.value}');
+        _name.sink.add(current.value.toString());
+
+        if (_puppy.name != current.value.toString()) {
+          emitUpdatingFields();
+        } else {
+          emitLoaded();
+        }
+      }
+    });
+
+    gender.onValueChanges(
+      onData: (previous, current) async* {
+        if (previous != current) {
+          // print('gender current value: ${current.value}');
+          _gender.sink.add(current.value ?? Gender.None);
+
+          if (_puppy.gender.toString().substring(7) !=
+              current.value.toString()) {
+
+            emitUpdatingFields();
+          } else {
+            emitLoaded();
+          }
+        }
+      },
+    );
+
+    breed.onValueChanges(
+      onData: (previous, current) async* {
+        if (previous != current) {
+          // print('breed current value: ${current.value}');
+          _breed.sink.add(current.value ?? BreedType.None);
+
+          if (_puppy.breedType != current.value) {
+            emitUpdatingFields();
+          } else {
+            emitLoaded();
+          }
+        }
+      },
+    );
+
+    characteristics.onValueChanges(
+      onData: (previous, current) async* {
+        if (previous != current) {
+          // print('displayCharacteristics current value: ${current.value}');
+          _characteristics.sink.add(current.value.toString());
+
+          if (_puppy.displayCharacteristics != current.value) {
+            emitUpdatingFields();
+          } else {
+            emitLoaded();
+          }
+        }
+      },
+    );
+
+  }
+
+  // CoordinatorBloc _coordinatorBloc;
+  final Puppy _puppy;
+  final PuppiesRepository repository;
+  static const int _maxCharacteristicsLength = 250;
+  static const int _maxNameLength = 30;
+
+  final _asset = BehaviorSubject<String>();
+  final _name = BehaviorSubject<String>();
+  final _gender = BehaviorSubject<Gender>();
+  final _characteristics = BehaviorSubject<String>();
+  final _breed = BehaviorSubject<BreedType>();
+
+  Stream<String> get asset$ => _asset.stream.startWith(_puppy.asset);
+
+  Stream<String> get name$ => _name.stream.startWith(_puppy.name);
+
+  Stream<Gender> get gender$ => _gender.stream.startWith(_puppy.gender);
+
+  Stream<String> get characteristics$ =>
+      _characteristics.stream.startWith(_puppy.displayCharacteristics ?? '');
+
+  Stream<BreedType> get breed$ => _breed.stream.startWith(_puppy.breedType);
+
+  Stream<bool> get isFormValid$ => Rx.combineLatest5(
```

Comment:
@StanevPrime I am not sure here if I should use streams like that or if I should return a state with a boolean if the save icon should be enabled when some of the fields has been changed. Also I am using a StreamBuilder and not a BlocBuilder when isFormValid$ is changed. I could not find examples with custom events or states for the form_bloc.